alibi_detect/saving/saving.py

```diff
@@ -503,3 +516,22 @@ def _save_kernel_config(kernel: Callable,
         cfg_kernel['src'], _ = _serialize_object(kernel_class, base_path, local_path.joinpath('kernel'))
 
     return cfg_kernel
+
+
+def _save_optimizer_config(optimizer: Union['tf.keras.optimizers.Optimizer', type]) -> dict:
+    """
+    Function to save tensorflow or pytorch optimizers.
+
+    Parameters
+    ----------
+    optimizer
+        The optimizer to save.
+
+    Returns
+    -------
+    Optimizer config dict.
+    """
+    if isinstance(optimizer, type):
+        return {'class_name': optimizer.__name__}
+    else:
+        return save_optimizer_config_tf(optimizer)
```

Comment:
If `optimizer` is an object, it must be a tensorflow optimizer, serialize it into a tensorflow conifg dict.